packages/astro/src/assets/build/generate.ts

```diff
@@ -103,9 +104,13 @@ export async function generateImage(
 
 	// If the image is local, we can just read it directly, otherwise we need to download it
 	if (isLocalImage) {
+		const outDir = isServerLikeOutput(config)
+			? config.build.server
+			: getOutDirWithinCwd(config.outDir);
+
 		imageData = await fs.promises.readFile(
 			new URL(
-				'.' + prependForwardSlash(join(config.build.assets, basename(originalImagePath))),
+				prependForwardSlash(join(outDir.pathname + config.build.assets, basename(originalImagePath))),
 				serverRoot
```

Comment:
We can put the `outDir.pathname` here as that doesn't work on Windows. Updating `serverRoot` to the above suggestion should do the trick.